Feature: Implement DAO class, API

CeNTePeDe opened the pull request.

CeNTePeDe committed:
removed list_streams
add constant variables
created:
- get_data_stream
Pydentic models:
- StreamerIn
- StreamerOut

CeNTePeDe committed:
created:
- router to streamer
- exception(InvalidIdInputError)
- created DAO

slavadan commented on core/constant_variables.py:
```diff
@@ -1,15 +1,23 @@
+HEADERS_TWITCH: dict = {
+    "Client-ID": os.environ.get("ID"),
+    "Authorization": os.environ.get("TOKEN"),
+}
```
ты же их получаешь в настройках кстати

slavadan replied:
не поправила

slavadan commented on parsers/parse_streamer.py:
```diff
@@ -0,0 +1,22 @@
+    data_streamers = response.json()["data"]
+    logger.info("get_data_stream is finished successfully")
+    logger.info(f"list_streamers is {data_streamers}")
+    return data_streamers
```
если у тебя используется только BASE_URL может не стоит его тогда передавать как параметр, тогда и райзить ничего не придеться

slavadan commented on database/connection_streamers.py:
```diff
@@ -0,0 +1,58 @@
+            {"id": id}, {"$set": streamer_data.dict()}
+        )
+        if streamer_update.modified_count == 0:
+            raise InvalidIdInputError(id=id)
```
не надо райзить ошибку если ничего не поменялось, зачем? в другом дао тыже не райзишь, чего ты все усложнаешь

slavadan commented on database/connection_streamers.py:
```diff
@@ -0,0 +1,58 @@
+    def get_item(self, id: str) -> StreamerIn:
+        streamer_data = self.collection.find_one({"id": id})
+        if streamer_data is None:
+            raise InvalidIdInputError(id=id)
```
не надо райзить ошибку если ничего не поменялось, зачем? в другом дао тыже не райзишь, чего ты все усложнаешь

slavadan commented on database/connection_streamers.py:
```diff
@@ -0,0 +1,58 @@
+        else:
+            updated_streamer = self.update_item(streamer_dict["id"], streamer_data)
+            logger.info(f"updated streamer {updated_streamer}")
+            return updated_streamer
```
порабы уже начать замечать, когда else уместен а когда нет. Как думаешь почему тут else не нужен

CeNTePeDe committed:
added
- id to url
- added Exception
removed const variables
removed url
removed:
- InvalidIdInputError

slavadan commented on database/connection_streamers.py:
```diff
@@ -0,0 +1,58 @@
+        )
+        if streamer_update is None:
+            raise HTTPException(status_code=404, detail="Streamer not found")
+        return streamer_update.modified_count
```
Оля ну вот опять ты сюда raise добавила. Тут шо медом намазано?

slavadan commented on database/connection_streamers.py:
```diff
@@ -0,0 +1,58 @@
+    def get_item(self, id: str) -> StreamerIn:
+        streamer_data = self.collection.find_one({"id": id})
+        if streamer_data is None:
+            raise HTTPException(status_code=404, detail="Streamer not found")
```
Оля ну вот опять ты сюда raise добавила. Тут шо медом намазано?

slavadan commented on parsers/parse_streamer.py:
```diff
@@ -0,0 +1,20 @@
+
+
+def get_data_stream(offset: Optional[int] = None) -> list[dict]:
+    from main import settings
```
все еще не пофикслила. Пофикси ты уже эти импорты. Можешь попроббовать поменять структуру проекта, только согласуй со мной.

slavadan commented on routers/streamer_routers.py:
```diff
@@ -0,0 +1,47 @@
+
+logger = logging.getLogger(__name__)
+
+streamer_routers = APIRouter()
+streamer_dao = StreamerDAO()
```
доп задание, сделать так чтобы не надо было создавать dao в каждом классе/функции а то так слишком  много конекшенов к бд создаваться будет. Почитай про lazy инициализацию ее плюсы и минусы: мб сюда ее тоже можно припаять

CeNTePeDe committed:
added
- added exception
- return None to update function

slavadan commented on database/connection_streamers.py:
```diff
@@ -40,17 +40,17 @@ def sort_item(self) -> list[StreamerOut]:
         )
         if streamer_update is None:
-            raise HTTPException(status_code=404, detail="Streamer not found")
+            return None
```
ну тут можно было бы и 0 возвращать, но да ладно

slavadan commented on routers/streamer_routers.py:
```diff
@@ -39,9 +42,14 @@ async def create_streamer(streamer: StreamerIn) -> int:
-    return streamer_dao.update_item(id=id, streamer_data=streamer_data)
+    streamer_updated = streamer_dao.update_item(id=id, streamer_data=streamer_data)
+    if streamer_updated == 0:
+        raise HTTPException(status_code=404, detail="Streamer not found")
```
а если он None вернет

CeNTePeDe committed:
Merge branch 'develop' into feature/api_dao
